### DISCUSSION TITLE
Can you run programmes from a phone

### DISCUSSION THREAD

**Discussion by benredstar** [pybricks/support#1363] [Q&A] [answered]
Hello all

I currently use the LEGO PoweredUp app to run control panels for my models such as this: https://www.instagram.com/p/C1j30l-MAMW/

I am wondering if I can use PyBricks instead. I understand PyBricks can create a program that can be uploaded to the hub directly; but I don't want to do that - I want to run the program from a phone and press buttons on a control panel to activate the different functions. Does PyBricks let you make control panels etc?

Many thanks

/ben

**Comment by laurensvalk** [comment, marked as answer]
> I want to run the program from a phone and press buttons on a control panel to activate the different functions

We don't currently have an editor with "control widgets" as you may have used in the Powered Up app.

But if this has enough buttons for your taste, you could use this one. You could also use combinations of buttons for even more options.

![image](https://github.com/pybricks/support/assets/12326241/8ff5d9b4-4f41-44f1-8c94-9c8a7c8e6039)

Or add some of your own logic to change what each button does when you press the center button, for example.

Physical buttons can be quite useful for stuff like this.

**Reply by benredstar** [reply]
Thanks for your reply. In this case the controller isn't really useful, I need more buttons than that.

I've used the sBrickPro app in the past; which was brilliant and let you construct your control panel with useful labels and so forth, such as the below. 

![Screenshot 2024-01-02 134337](https://github.com/pybricks/support/assets/55100089/9792189d-7bbe-4a44-8bd0-88a8d2009e9a)

However support for this seems to have been abandoned by sBrick and I've switched to PoweredUp. If PyBricks started to let you do this kind of stuff I'd switch in a heartbeat, but until then I'll have to stick to PoweredUp.

/ben

**Reply by laurensvalk** [reply]
You could make this relatively easily by using the middle button to toggle between drive/aim/shoot, and using the remote led to show you which mode is active.

**Reply by benredstar** [reply]
mmm that is appealing; plus the remote controls are cheaper than phones...

/b

**Reply by benredstar** [reply]
@laurensvalk I did some more thinking about this.

What if there was an Android app that simply lets you make control panels, buttons, sliders, add labels, that sort of stuff. If that app could then communicate the outputs of its controls via bluetooth to the LEGO hub (running a PyBricks program), you'd have a way of replacing the hardware LEGO controller with a phone. The software is still actually running on the LEGO hub, it's just communicating with the android app to read the outputs of buttons.

Such an app might be very simple to engineer, because it's not actually DOING anything; just presenting an interface and relaying the status of the buttons. It's even possible there's an existing app out there that can be used to do this and all you'd need to do is get it to talk to the LEGO hub and Pybricks.

/ben

**Reply by laurensvalk** [reply]
Yes, in fact it might be a nice widget to have within Pybricks Code.

**Reply by benredstar** [reply]
Do you know if there's way on this forum to suggest the idea to the devs?

/b
